src/lib.rs

```diff
@@ -118,13 +194,16 @@ impl Mapping {
     ///
     /// If the `Mapping` was constructed with `with_functions`, information about the containing
     /// function may also be returned when available.
-    pub fn locate(&self, addr: u64) -> Option<(path::PathBuf, u64, Option<Cow<str>>)> {
-        self.inner.locate(addr)
+    pub fn locate(&self, addr: u64) -> Option<(path::PathBuf, Option<u64>, Option<Cow<str>>)> {
+        self.inner.locate(addr).unwrap_or_else(|_| {
+            // TODO: we probably want to log this, or assert failure when testing?
```

Comment:
Why not return the error and propagate it up the stack where the caller can decide what is appropriate?